core/src/main/scala/kafka/zk/KafkaZkClient.scala

```diff
@@ -872,6 +906,14 @@ class KafkaZkClient(zooKeeperClient: ZooKeeperClient, isSecure: Boolean, time: T
     retryRequestUntilConnected(deleteRequest)
   }
 
+  /**
+   * Checks the controller path existence
+   * @return true if controller exists else false
+   */
+  def controllerPathExists(): Boolean = {
+    pathExists(ControllerZNode.path)
```

Comment:
Can we make pathExists method public? So that the test classes can call directly.

Reply:
Are there more cases where we need it or are they all covered in this PR?

Reply:
There are couple of more cases of zkUtils.pathExists usage. not sure about adding these path check to zkClient.  In future we may add more zk paths. so thought of making pathExists public.

TestUtils.secureZkPaths/verifySecureZkAcls -> zkUtils.pathExists(path)
TopicCommandTest - > zkUtils.pathExists(topicDeletePath)
ServerStartupTest  -> zkUtils.pathExists(zookeeperChroot)

Reply:
I don't see the harm given we have already exposed methods like `makeSurePersistentPathExists` and `createTopLevelPaths`.